hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metrics/datadog/DatadogHttpClient.java

```diff
@@ -84,8 +84,8 @@ private void validateApiKey() {
     try (CloseableHttpResponse response = client.execute(request)) {
       int statusCode = response.getStatusLine().getStatusCode();
       ValidationUtils.checkState(statusCode == HttpStatus.SC_OK, "API key is invalid.");
-    } catch (IOException e) {
-      throw new IllegalStateException("Failed to connect to Datadog to validate API key.", e);
+    } catch (IOException | IllegalStateException e) {
+      LOG.warn(String.format("Failed to connect to Datadog to validate API key. %s", e.getMessage()));
```

Comment:
do you mean, catch any exeption ? this makes sense

Reply:
> Should we still fail the job if the metrics collector does not work?

+1

Reply:
What I meant is, we should not change the behavior of throwing an exception here.  If the metric collection does not work due to API key, it should fail the job so that the user knows and fixes it before proceeding.

Reply:
I don't understand the rational here.  Should user know the API key is not working and fix it before running the job again to properly generate the metrics?  It's not a good idea to silently fail here.

Reply:
Got it.  I suggest having a feature flag on whether to fail the job if the metric provider does not work.  By default, it's on, i.e., failing the job due to metric provider, the same behavior as before, while users can turn this off in the case metrics can be skipped.

Reply:
Sure, will do that

Reply:
https://hudi.apache.org/docs/configurations#hoodiemetricsdatadogapikeyskipvalidation
shame on me, the option to skip validation already exist. Then this PR is useless.